### PR TITLE
Reformat classes to use dependency injection.

### PR DESCRIPTION
**Pull request by st55774**
This pull request adds Guice dependency injection to the project.
Reformat classes to services that will be transformed into singletons.
Remove static contents and using dependency injections.
Create a service for managing password database files.

### CODE REVIEW

File: src/main/java/cz/upce/fei/inptp/zz/entity/PasswordDatabase.java [hunk 1]
```diff
@@ -18,20 +18,12 @@
     
     private List<Password> passwords;
 
-    public PasswordDatabase(File file, String passwd) {
-        this.file = file;
+    public PasswordDatabase(String file, String passwd, List<Password> passwords) {
```
Comment (rodi0878) [inline src/main/java/cz/upce/fei/inptp/zz/entity/PasswordDatabase.java]: This is wrong, File is prefered Java way of abstracting file/folder paths

File: src/main/java/cz/upce/fei/inptp/zz/injector/InstanceInjector.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import cz.upce.fei.inptp.zz.service.crypto.CryptoFileServiceModule;
+import cz.upce.fei.inptp.zz.service.json.JSONFileServiceModule;
+import cz.upce.fei.inptp.zz.service.password.PasswordDatabaseServiceModule;
+
+/**
+ * It provides the dependency injector for project.
+ *
+ * @author Ondřej Chrbolka
```
Comment (rodi0878) [inline src/main/java/cz/upce/fei/inptp/zz/injector/InstanceInjector.java]: no authors comments - for this we have commit history

File: src/main/java/cz/upce/fei/inptp/zz/service/password/JSONPasswordDatabaseService.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    private CryptoService cryptoService;
+
+    @Inject
+    public JSONPasswordDatabaseService(CryptoService cryptoService) {
+        this.cryptoService = cryptoService;
+    }
+
+    @Override
+    public PasswordDatabase openPasswordDatabase(String path, final String password) {
```
Comment (rodi0878) [inline src/main/java/cz/upce/fei/inptp/zz/service/password/JSONPasswordDatabaseService.java]: why final password?

File: src/main/java/cz/upce/fei/inptp/zz/service/password/JSONPasswordDatabaseService.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        String content = cryptoService.readFile(new File(path), password);
```
Comment (rodi0878) [inline src/main/java/cz/upce/fei/inptp/zz/service/password/JSONPasswordDatabaseService.java]: new File(...) this also means that you've chosen wrong abstraction and you should stay with File

File: src/main/java/cz/upce/fei/inptp/zz/service/password/JSONPasswordDatabaseService.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+        List<Password> passwordFromJSON = new JSONFileService().fromJson(content);
```
Comment (rodi0878) [inline src/main/java/cz/upce/fei/inptp/zz/service/password/JSONPasswordDatabaseService.java]: no DI here?

File: src/main/java/cz/upce/fei/inptp/zz/injector/InstanceInjector.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    private InstanceInjector(){
+    }
+
+    /**
+     * It will return the single instance of dependency injector.
+     *
+     * @return dependency injector.
+     * */
+    public static Injector injector(){
+        if(injector == null){
```
Comment (rodi0878) [inline src/main/java/cz/upce/fei/inptp/zz/injector/InstanceInjector.java]: possible threading problem, do not use lazy initialization

**Review by rodi0878 — CHANGES_REQUESTED** [review]
Fine idea but current state is not ready for merging. Please look at my comments and adapt patch to current code state

**Comment by st55774** [comment]
@rodi0878 
thank you for your feedback. I fix all issues you mentioned in the comments and resolved merge conflicts. I hope it is all right now. If no, please let me know.

File: src/main/java/cz/upce/fei/inptp/zz/injector/InstanceInjector.java (hunk 6 above)
Comment (st55774) [reply src/main/java/cz/upce/fei/inptp/zz/injector/InstanceInjector.java]: Sorry about that. We used this method in OOP. I was searching on the internet how to patch it and I found this link (https://stackoverflow.com/questions/8297705/how-to-implement-thread-safe-lazy-initialization) where the author had the same problem. So I used the correct solution to fix.

File: src/main/java/cz/upce/fei/inptp/zz/service/password/JSONPasswordDatabaseService.java (hunk 5 above)
Comment (st55774) [reply src/main/java/cz/upce/fei/inptp/zz/service/password/JSONPasswordDatabaseService.java]: I forgot. Too much refactoring in the other classes.

File: src/main/java/cz/upce/fei/inptp/zz/service/password/JSONPasswordDatabaseService.java (hunk 4 above)
Comment (st55774) [reply src/main/java/cz/upce/fei/inptp/zz/service/password/JSONPasswordDatabaseService.java]: I will fix.

File: src/main/java/cz/upce/fei/inptp/zz/service/password/JSONPasswordDatabaseService.java (hunk 3 above)
Comment (st55774) [reply src/main/java/cz/upce/fei/inptp/zz/service/password/JSONPasswordDatabaseService.java]: It shouldn't. When I was using NetBeans IDE (version 8 I think, not sure). When you used the IDE tool to extract some part of code to a method, some parameters made IDE final... Not sure why (IntelliJ doesn't do that in default). Sometimes I forgot a will make parameters final.

File: src/main/java/cz/upce/fei/inptp/zz/injector/InstanceInjector.java (hunk 2 above)
Comment (st55774) [reply src/main/java/cz/upce/fei/inptp/zz/injector/InstanceInjector.java]: Removed.

File: src/main/java/cz/upce/fei/inptp/zz/entity/PasswordDatabase.java (hunk 1 above)
Comment (st55774) [reply src/main/java/cz/upce/fei/inptp/zz/entity/PasswordDatabase.java]: Ok, I was thinking to use java.nio.Path but a lot of methods still use java.io.File so I thought that String will be better. I will fix.

**Pull request merged**